crates/oxc_linter/src/config/config_store.rs

```diff
@@ -156,37 +156,32 @@ impl Config {
 
         let mut env = self.base.config.env.clone();
         let mut globals = self.base.config.globals.clone();
-        let mut plugins = self.base.config.plugins.clone();
+        let mut plugins = self.base.config.plugins;
 
         for override_config in overrides_to_apply.clone() {
-            if let Some(override_plugins) = &override_config.plugins {
-                plugins.builtin |= override_plugins.builtin;
-                for p in &override_plugins.external {
-                    plugins.external.insert(p.clone());
-                }
+            if let Some(override_plugins) = override_config.plugins {
+                plugins |= override_plugins;
```

Comment:
I *think* this is OK. We're not storing external plugins in `ResolvedLinterState`, but I don't think we need to, because all that matters at this point is which external *rules* are enabled.